docs/tutorials/pytorch.ipynb

```diff
@@ -0,0 +1,507 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3bddce3b-ed2f-4a5c-b3c6-2f1a3a51c2d9",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def evaluate(dataloader):\n",
+    "    model.eval()\n",
+    "    correct = 0\n",
+    "    with torch.no_grad():\n",
+    "        for batch in dataloader:\n",
+    "            x = batch['image'].to(device)\n",
+    "            y = batch['label'].to(device)\n",
+    "\n",
+    "            # Forward pass\n",
+    "            y_hat = model(x)\n",
+    "            correct += (y_hat.argmax(1) == y).type(torch.float).sum().item()\n",
+    "\n",
+    "    correct /= len(dataloader.dataset)\n",
+    "    print(f'Accuracy: {correct:.0%}')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f62a54e5-897d-476c-8d84-381993dbabbd",
+   "metadata": {},
+   "source": [
+    "## Putting It All Together\n",
+    "\n",
+    "In machine learning, we typically iterate over our datasets multiple times. Each full pass through the dataset is called an *epoch*. The following hyperparameter controls the number of epoch for which we train our model, and can be modified to train the model for longer:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "eb5dc7e8-6cb3-4457-83ad-7fa5aef8ea0c",
+   "metadata": {
+    "nbmake": {
+     "mock": {
+      "epochs": 1
+     }
+    }
+   },
+   "outputs": [],
+   "source": [
+    "epochs = 100"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f53526e6-54a3-43f7-a377-dca298730387",
+   "metadata": {},
+   "source": [
+    "During each epoch, we train the model on our training dataset, then evaluate its performance on the validation dataset. The goal is for training loss to decrease and validation accuracy to increase, although you should expect noise in the training process. Generally, you want to train the model until the validation accuracy starts to plateau or even decrease."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "97601568-ba75-443d-81cf-494956b2924c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "for epoch in range(epochs):\n",
+    "    print(f'Epoch: {epoch}')\n",
+    "    train(train_dataloader)\n",
+    "    evaluate(val_dataloader)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e130fc89-0823-4814-85f8-d4416d6df395",
+   "metadata": {},
+   "source": [
+    "Finally, we evaluate our performance on the test dataset. Note that we are only training our model on a toy dataset consisting of 100 images. If we instead trained on the full dataset (replace `EuroSAT100` with `EuroSAT` in the above code), we would likely get much higher performance."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "7cd0bd25-e19a-4b26-94a1-fe9a544e8afd",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "evaluate(test_dataloader)"
```

Comment:
Final accuracies are 65% on validation and 85% on testing. Pretty good for only 100 images. We are nowhere near plateauing, and could run for longer to get higher accuracies, but I think this proves the point.